packages/docusaurus-plugin-content-blog/src/__tests__/blogUtils.test.ts

```diff
@@ -295,3 +297,49 @@ describe('linkify', () => {
     } as BlogBrokenMarkdownLink);
   });
 });
+
+const defaultValuesForOtherKeys: Omit<BlogPostMetadata, 'date'> = {
+  source: '',
+  title: '',
+  formattedDate: '',
+  permalink: '',
+  description: '',
+  hasTruncateMarker: false,
+  authors: [],
+  frontMatter: {},
+  tags: [],
+  unlisted: false,
+};
+const createBlogPost = (args: Partial<BlogPost>): BlogPost => ({
+  id: '',
+  metadata: {
+    date: new Date(),
+    ...defaultValuesForOtherKeys,
+    ...args.metadata,
+  },
+  content: args.content || '',
+});
+
+describe('processBlogPosts', () => {
+  const blogPost2022 = createBlogPost({
+    metadata: {date: new Date('2022-01-01'), ...defaultValuesForOtherKeys},
+  });
+  const blogPost2023 = createBlogPost({
+    metadata: {date: new Date('2023-01-01'), ...defaultValuesForOtherKeys},
+  });
+  const blogPost2024 = createBlogPost({
+    metadata: {date: new Date('2024-01-01'), ...defaultValuesForOtherKeys},
+  });
+
+  it('filter blogs', async () => {
```

Comment:
we should also test other cases, notably imperarive modification of the input list (`blogPosts.sort(comparator)` for example)